### PR TITLE
Remove statistics#download_counter

### PR DESCRIPTION
**Pull request by dmarcoux**
With the changes from 17cd6b1, this action has been broken for more than 9 years now. It's quite safe to assume we can remove this code. 

There's a data migration to clean up production.

### CODE REVIEW

File: src/api/test/functional/backend_test.rb [hunk 1]
```diff
@@ -13,8 +13,8 @@ def test_validate_bsxml
       # map schema names
       if ['about', 'activity', 'added_timestamp', 'announcement', 'announcements', 'new_announcement', 'architecture', 'attrib', 'attrib_type',
           'attrib_namespace', 'attribute_namespace_meta', 'collection_objects_by_tag',
-          'collection_objects_with_tags_by_user', 'configuration', 'directory_filelist', 'directory_view', 'download_counter',
-          'download_counter_summary', 'download_stats', 'group', 'highest_rated', 'issue_tracker',
+          'collection_objects_with_tags_by_user', 'configuration', 'directory_filelist', 'directory_view',
+          'download_stats', 'group', 'highest_rated', 'issue_tracker',
```
Comment (coolo) [inline src/api/test/functional/backend_test.rb:17]: btw: the download_stats DTD is just as dead.

Comment (dmarcoux) [reply src/api/test/functional/backend_test.rb:17]: Thanks for the hint!

**Review by vpereira — APPROVED** [review]

**Comment by vpereira** [comment]
please rebase @dmarcoux

**Comment by dmarcoux** [comment]
@vpereira Done

**Comment by hennevogel** [comment]
You need to mention things in the ReleaseNotes

**Pull request merged**